### DISCUSSION TITLE
Configuring a services HttpClient per tenant

### DISCUSSION THREAD

**Discussion by chihabhajji** [Finbuckle/Finbuckle.MultiTenant#561] [Q&A] [answered]
Hello, i was wondering what is the best way to go about setting up a services HttpClient based on the current tenant info, i cant seem to use PerTenantOptions or named options because there are no httpclientoptions

**Comment by AndrewTriesToCode** [comment, marked as answer]
Hi, the best method I've found it to use your own Options class with whatever settings you need in combination with a custom HttpClientHandler implementation which makes use of your Options class.

Info on HttpClientHandler:
https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler?view=net-6.0

 I would register it as a transient and then register HttpClient with a factory method that instantiates the HttpClient with the HttpClientHandler. I hope this is helpful.

**Comment by chihabhajji** [comment]
using options sounds like a better way than what i'm currently doing 
`     public GateXRoadClient(FSHTenantInfo? tenantInfo, IConfiguration configuration) : base(Client)
    {
        if (tenantInfo?.XRoadSettings is not null && !string.IsNullOrEmpty(tenantInfo.XRoadSettings.SecurityServerToken) && !string.IsNullOrEmpty(tenantInfo.XRoadSettings.SecurityServerUrl))
        {
            BaseUrl = tenantInfo.XRoadSettings.SecurityServerUrl;
            ApiKey = tenantInfo.XRoadSettings.SecurityServerToken;
            Client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("X-Road-ApiKey", $"token={tenantInfo.XRoadSettings.SecurityServerToken}");
        }
        else
        {
            if (tenantInfo?.Id == MultitenancyConstants.Root.Id)
            {
                var ctx = configuration.GetSection(nameof(XRoadSettings)).Get<XRoadSettings>();
                Client.DefaultRequestHeaders.Authorization = ctx.AuthenticationHeaderValue;
                BaseUrl = ctx.SSUrl;
                ApiKey = ctx.SSToken;
            }
            else
            {
                throw new UnauthorizedException("Unauthorized access to XROAD");
            }
        }
    }
`
thank you for the suggestion
